src/main/java/spoon/reflect/declaration/CtFormalTypeDeclarer.java

```diff
@@ -43,4 +45,9 @@
 	 * Removes a type parameters from this generic element.
 	 */
 	boolean removeFormalCtTypeParameter(CtTypeParameter formalTypeParameter);
+
+	/**
+	 * @return the instance of {@link GenericTypeAdapter} which can adapt type parameters from different scopes to scope of this declarer
+	 */
+	GenericTypeAdapter getGenericTypeAdapter();
```

Comment:
really hard to understand, I wouldn't put it here. Is it required?

If we remove it, we won't have to touch all the classes implementing it, and this simplifies the PR much.